utils/src/com/cloud/utils/net/NetUtils.java

```diff
@@ -329,35 +329,46 @@ public static String getMacAddress(InetAddress address) {
         return sb.toString();
     }
 
-    public static long getMacAddressAsLong(InetAddress address) {
+    public static long getMacAddressAsLong(final InetAddress address) {
         long macAddressAsLong = 0;
         try {
-            NetworkInterface ni = NetworkInterface.getByInetAddress(address);
-            byte[] mac = ni.getHardwareAddress();
+            final NetworkInterface ni = NetworkInterface.getByInetAddress(address);
+            final byte[] mac = ni.getHardwareAddress();
 
             for (int i = 0; i < mac.length; i++) {
-                macAddressAsLong |= ((long)(mac[i] & 0xff) << (mac.length - i - 1) * 8);
+                macAddressAsLong |= (long)(mac[i] & 0xff) << (mac.length - i - 1) * 8;
             }
 
-        } catch (SocketException e) {
+        } catch (final SocketException e) {
             s_logger.error("SocketException when trying to retrieve MAC address", e);
         }
 
         return macAddressAsLong;
     }
 
-    public static boolean ipRangesOverlap(String startIp1, String endIp1, String startIp2, String endIp2) {
-        long startIp1Long = ip2Long(startIp1);
+    public static boolean ipRangesOverlap(final String startIp1, final String endIp1, final String startIp2, final String endIp2) {
+        final long startIp1Long = ip2Long(startIp1);
         long endIp1Long = startIp1Long;
         if (endIp1 != null) {
             endIp1Long = ip2Long(endIp1);
         }
-        long startIp2Long = ip2Long(startIp2);
+        final long startIp2Long = ip2Long(startIp2);
         long endIp2Long = startIp2Long;
         if (endIp2 != null) {
             endIp2Long = ip2Long(endIp2);
         }
 
+        // check if the gatewayip is the part of the ip range being added.
+        // RFC 3021 - 31-Bit Prefixes on IPv4 Point-to-Point Links
+        //     GW              Netmask         Stat IP        End IP
+        // 192.168.24.0 - 255.255.255.254 - 192.168.24.0 - 192.168.24.1
+        // https://tools.ietf.org/html/rfc3021
+        // Added by Wilder Rodrigues
+        final int ip31bitPrefixOffSet = 1;
+        if (startIp2Long - startIp1Long == startIp2Long - (endIp1Long - ip31bitPrefixOffSet)) {
```

Comment:
No. It doesn't because the isIpWithtInCidrRange() is checked before, in the ConfigurationManagerImpl.

As you can see, the method you commented on doesn't check the CIDR.